does not compile with gcc 10

## Prerequisites
These are MANDATORY, otherwise the issue will be automatically closed.

<!-- Fill with an 'x'. -->
* [x] I agree to fill this issue template.
* [x] I have read the [Troubleshooting Guide] and [Support Instructions].

[Troubleshooting Guide]: https://doc-kurento.readthedocs.io/en/latest/user/troubleshooting.html
[Support Instructions]: https://github.com/Kurento/.github/blob/master/SUPPORT.md

## Issue description
Trying to compile `kms-core` after my distribution updated its default to gcc version 10 fails.

## Context
```
In file included from /build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:21:
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:34:34: error: 'std::string' has not been declared
   34 |   static void registerType (std::string type,
      |                                  ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:37:26: error: expected ';' at end of member declaration
   37 |   static RegisterParent *createRegister (std::string type);
      |                          ^~~~~~~~~~~~~~
      |                                        ;
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:37:53: error: expected ')' before 'type'
   37 |   static RegisterParent *createRegister (std::string type);
      |                                         ~           ^~~~~
      |                                                     )
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:25: error: 'string' is not a member of 'std'
   40 |   static std::map <std::string, std::function<RegisterParent* (void) >>
      |                         ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:24:1: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
   23 | #include <atomic>
  +++ |+#include <string>
   24 |
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:25: error: 'string' is not a member of 'std'
   40 |   static std::map <std::string, std::function<RegisterParent* (void) >>
      |                         ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:25: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:70: error: template argument 1 is invalid
   40 |   static std::map <std::string, std::function<RegisterParent* (void) >>
      |                                                                      ^~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:70: error: template argument 3 is invalid
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.hpp:40:70: error: template argument 4 is invalid
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:16: error: 'string' is not a member of 'std'
   26 | std::map <std::string, std::function<RegisterParent* (void) >>
      |                ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:22:1: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
   21 | #include "RegisterParent.hpp"
  +++ |+#include <string>
   22 |
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:16: error: 'string' is not a member of 'std'
   26 | std::map <std::string, std::function<RegisterParent* (void) >>
      |                ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:16: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:61: error: template argument 1 is invalid
   26 | std::map <std::string, std::function<RegisterParent* (void) >>
      |                                                             ^~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:61: error: template argument 3 is invalid
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:26:61: error: template argument 4 is invalid
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:30:1: error: variable or field 'registerType' declared void
   30 | RegisterParent::registerType (std::string type,
      | ^~~~~~~~~~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:30:36: error: 'string' is not a member of 'std'
   30 | RegisterParent::registerType (std::string type,
      |                                    ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:30:36: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:31:70: error: expected primary-expression before 'func'
   31 |                               std::function<RegisterParent* (void) > func)
      |                                                                      ^~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:37:38: error: 'string' is not a member of 'std'
   37 | RegisterParent::createRegister (std::string type)
      |                                      ^~~~~~
/build/kms-core-source-6.15.0/src/server/implementation/RegisterParent.cpp:37:38: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
make[2]: *** [src/server/CMakeFiles/kmscoreimpl.dir/build.make:246: src/server/CMakeFiles/kmscoreimpl.dir/implementation/RegisterParent.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:318: src/server/CMakeFiles/kmscoreimpl.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

## (Optional) Possible solution
For now I can just override it back to gcc 9.

### About Kurento Media Server
* Kurento version: 6.15.0
* Server OS:         NixOS unstable
* Installation method:
    <!-- Fill with an 'x' in the boxes that apply. -->
  - [ ] [apt-get]
  - [ ] [Docker]
  - [ ] [AWS CloudFormation]
  - [x] [Built from sources]

[apt-get]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-local
[Docker]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-docker
[AWS CloudFormation]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-aws
[Built from sources]: https://doc-kurento.readthedocs.io/en/latest/dev/dev_guide.html#dev-sources

--- a/src/server/implementation/RegisterParent.hpp
+++ b/src/server/implementation/RegisterParent.hpp
@@ -21,6 +21,7 @@
 #include <map>
 #include <functional>
 #include <atomic>
+#include <string>
 
 namespace kurento
 {

